From 9ccce877a9cc61163d4b78ac62b185f86a25e14c Mon Sep 17 00:00:00 2001
From: Subhankar Paul <subhankar.paul@intel.com>
Date: Thu, 23 Jul 2015 14:14:25 +0530
Subject: [PATCH] ASoC: Intel: Readjusting vtsv net/grammar file location in
 memory.

Memory arrangement was wrong for net and grammar file, so there
are cache miss happening while running single phrase LPAL.
Rearrangement of memory has been in accordance with LPE firmware
to avoid cache miss. Now max allowed vtsv_net file size 368Kb
and vtsv_grammar file size 16Kb.

Change-Id: Ie2802626c15d4494695cfe3537f8701a7a44c99a
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-42128
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-14952
Signed-off-by: Subhankar Paul <subhankar.paul@intel.com>
---
 sound/soc/intel/sst/sst_stream.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/intel/sst/sst_stream.c b/sound/soc/intel/sst/sst_stream.c
index 3d1fd7cf356d..d194ddb05b28 100644
--- a/sound/soc/intel/sst/sst_stream.c
+++ b/sound/soc/intel/sst/sst_stream.c
@@ -809,8 +809,8 @@ int sst_format_vtsv_message(struct intel_sst_drv *ctx,
 }
 
 /* Total 384 KB is allocated for vtsv net/grammar file */
-#define VTSV_NET_MAX_SIZE 0x58000	/* max net file size: 352 KB */
-#define VTSV_GRAMMAR_MAX_SIZE 0x8000 /* max grammar file size: 32 KB */
+#define VTSV_NET_MAX_SIZE 0x5C000	/* max net file size: 368 KB */
+#define VTSV_GRAMMAR_MAX_SIZE 0x4000 /* max grammar file size: 16 KB */
 
 int sst_cache_vtsv_libs(struct intel_sst_drv *ctx)
 {
-- 
1.9.1

